fe/fe-core/src/main/java/com/starrocks/analysis/TimeSliceTypeExpr.java

```diff
@@ -0,0 +1,126 @@
+// This file is made available under Elastic License 2.0.
+// This file is based on code available under the Apache license here:
+//   https://github.com/apache/incubator-doris/blob/master/fe/fe-core/src/main/java/org/apache/doris/analysis/TimeSliceTypeExpr.java
+
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package com.starrocks.analysis;
+
+import com.google.common.base.Preconditions;
+import com.starrocks.analysis.ArithmeticExpr.Operator;
+import com.starrocks.common.AnalysisException;
+import com.starrocks.sql.ast.AstVisitor;
+import com.starrocks.thrift.TExprNode;
+import com.starrocks.thrift.TExprNodeType;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Describes the addition and subtraction of time units from timestamps.
+ * Arithmetic expressions on timestamps are syntactic sugar.
+ * They are executed as function call exprs in the BE.
+ */
+public class TimeSliceTypeExpr extends Expr {
```

Comment:
why do we need to add TimeSliceTypeExpr?

Reply:
> why do we need to add TimeSliceTypeExpr?

Because TimestampArithmeticExpr is used for time functions with arithmetic, we should use another expr with time_slice function.

Reply:
I also feel there is no need to add TimeSliceExpr, using FunctionCallExpr is enough?

Reply:
> I also feel there is no need to add TimeSliceExpr, using FunctionCallExpr is enough?

Yes, I've remove and use FunctionCallExpr instead.